### PR TITLE
Add app.feeddeck.feeddeck

### PR DESCRIPTION
**Pull request by ricoberger**
### Please confirm your submission meets all the criteria

- [x] I have read the [App Requirements][reqs] and [App Maintenance][maint] pages.
- [x] My pull request follows the instructions at [App Submission][submission].
- [x] I am using only the minimal set of permissions. *(If not, please explain each non-standard permission.)*
- [x] All assets referenced in the manifest are redistributable by any party.  If not, the unredistributable parts are using an extra-data source type.
- [x] I am an upstream contributor to the project. If not, I contacted upstream developers about submitting their software to Flathub. **Link:**
- [x] I own the domain used in the application ID or the domain has a policy for delegating subdomains (e.g. GitHub, SourceForge).
- [x] Any additional patches or files have been submitted to the upstream projects concerned. *(If not, explain why.)*

[reqs]: https://docs.flathub.org/docs/for-app-authors/requirements
[maint]: https://docs.flathub.org/docs/for-app-authors/maintanance
[submission]: https://docs.flathub.org/docs/for-app-authors/submission

### CODE REVIEW

**Comment by ricoberger** [comment]
I know that it is not recommended to have all the [metadata files](https://docs.flathub.org/docs/for-app-authors/requirements/#acceptable-but-should-be-submitted-upstream) within this repository, but since it is also not forbidden I would like to have all Flatpak / Flathub specific files within one place. If this is a problem please let me know and I will create an additional repository for them.

File: app.feeddeck.feeddeck.yml [hunk 1]
```diff
@@ -0,0 +1,43 @@
+app-id: app.feeddeck.feeddeck
+runtime: org.freedesktop.Platform
+runtime-version: "22.08"
```
Comment (bbhtt) [inline app.feeddeck.feeddeck.yml:3]: Current runtime is 23.08. Try to use that one instead.

File: flathub.json
Comment (bbhtt) [inline flathub.json:1]: You don't need this, these are the only arches avaialble at the moment.

File: screenshots/1.png
Comment (bbhtt) [inline screenshots/1.png:1]: Do the screenshots need to be here too? Can you not put them in the website/repo?

**Comment by bbhtt** [comment]
> I know that it is not recommended to have all the [metadata files](https://docs.flathub.org/docs/for-app-authors/requirements/#acceptable-but-should-be-submitted-upstream) within this repository, but since it is also not forbidden I would like to have all Flatpak / Flathub specific files within one place. If this is a problem please let me know and I will create an additional repository for them.

You don't need to create a separate repo. Ideally it should be in the upstream repo since you have add release notes/dates, translations over time.

File: app.feeddeck.feeddeck.svg
Comment (bbhtt) [inline app.feeddeck.feeddeck.svg:1]: Does the icon needs to be here too?

File: app.feeddeck.feeddeck.desktop
Comment (bbhtt) [inline app.feeddeck.feeddeck.desktop:1]: You don't have a desktop file upstream?

**Comment by ricoberger** [comment]
Hi @bbhtt, thanks for the fast review. I added all the metadata files to the upstream repository.

File: app.feeddeck.feeddeck.yml [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+sdk: org.freedesktop.Sdk
+command: feeddeck
+
+finish-args:
+  - --share=ipc
+  - --socket=fallback-x11
+  - --socket=wayland
+  - --device=dri
+  - --share=network
+  - --filesystem=home
```
Comment (bbhtt) [inline app.feeddeck.feeddeck.yml:13]: Why do you need home access?

**Comment by bbhtt** [comment]
bot, build app.feeddeck.feeddeck

File: app.feeddeck.feeddeck.yml [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+      - install -Dm644 app.feeddeck.feeddeck.svg /app/share/icons/hicolor/scalable/apps/app.feeddeck.feeddeck.svg
+      - install -Dm644 app.feeddeck.feeddeck.desktop -t /app/share/applications/
+      - install -Dm644 app.feeddeck.feeddeck.metainfo.xml -t /app/share/metainfo/
+    sources:
+      - type: file
+        url: https://raw.githubusercontent.com/feeddeck/feeddeck/main/app/linux/flatpak/app.feeddeck.feeddeck.desktop
+        sha256: c66ab88c9683b69914e12d4bc780a46473631b2539ca83503fae404145eae7cf
+      - type: file
+        url: https://raw.githubusercontent.com/feeddeck/feeddeck/main/app/linux/flatpak/app.feeddeck.feeddeck.metainfo.xml
+        sha256: 6c818c36163ebffddea746bea2080bf6fbd571b5eb9f85a0074757b1106d1614
```
Comment (bbhtt) [inline app.feeddeck.feeddeck.yml:31]: Wrong checksum here

File: app.feeddeck.feeddeck.yml (hunk 2 above)
Comment (ricoberger) [reply app.feeddeck.feeddeck.yml:13]: The app uses Supabase, which stores the users information in the `home` directory. If this permission is not set, a users session is not persisted and the user must login again as soon as the app is restarted.

Comment (bbhtt) [reply app.feeddeck.feeddeck.yml:13]: > in the home directory

Where?

Sounds like it should use the standard `XDG_` variables for storing.

File: app.feeddeck.feeddeck.yml (hunk 3 above)
Comment (ricoberger) [reply app.feeddeck.feeddeck.yml]: This should be fixed in https://github.com/flathub/flathub/pull/4568/commits/701b65f69601c63c2b95bce065f991a59b2a9959

File: app.feeddeck.feeddeck.yml (hunk 2 above)
Comment (ricoberger) [reply app.feeddeck.feeddeck.yml:13]: It's stored in `~/Documents/auth`.

Should this be changed to `- --filesystem=xdg-documents`?

**Comment by hfiguiere** [comment]
> I would like to have all Flatpak / Flathub specific files within one place

The only flaptak specific file is the manifest. Everything else is standard and is NOT flatpak / flathub specific.

File: app.feeddeck.feeddeck.yml [hunk 4]
```diff
@@ -0,0 +1,46 @@
...
+      - type: file
+        url: https://raw.githubusercontent.com/feeddeck/feeddeck/main/app/linux/flatpak/app.feeddeck.feeddeck.metainfo.xml
+        sha256: 441fc4e08a6030b1e2238f93bced6e11f52da9d795d4a89c1609646609d61e16
+      - type: file
+        url: https://raw.githubusercontent.com/feeddeck/feeddeck/main/app/linux/flatpak/app.feeddeck.feeddeck.svg
+        sha256: 4a73d70cdaa6fe230eb802fe0f5456df46dfe399826b044086ad46b74c2bd6ce
+      - type: archive
+        only-arches:
+          - x86_64
+        url: https://github.com/feeddeck/feeddeck/releases/download/v1.0.0/feeddeck-linux-x86_64.tar.gz
```
Comment (hfiguiere) [inline app.feeddeck.feeddeck.yml:38]: why not building from source?

Comment (hfiguiere) [reply app.feeddeck.feeddeck.yml:38]: oh flutter. nevermind.

File: app.feeddeck.feeddeck.yml (hunk 2 above)
Comment (bbhtt) [reply app.feeddeck.feeddeck.yml:13]: Yea probably. Unless you need it for anything else.

Comment (ricoberger) [reply app.feeddeck.feeddeck.yml]: I changed it. I think it's not needed for sth. else, at least it is also working when trying it locally.

**Comment by bbhtt** [comment]
bot, build app.feeddeck.feeddeck

File: app.feeddeck.feeddeck.yml [hunk 5]
```diff
@@ -0,0 +1,46 @@
...
+        url: https://raw.githubusercontent.com/feeddeck/feeddeck/main/app/linux/flatpak/app.feeddeck.feeddeck.desktop
+        sha256: c66ab88c9683b69914e12d4bc780a46473631b2539ca83503fae404145eae7cf
+      - type: file
+        url: https://raw.githubusercontent.com/feeddeck/feeddeck/main/app/linux/flatpak/app.feeddeck.feeddeck.metainfo.xml
+        sha256: 441fc4e08a6030b1e2238f93bced6e11f52da9d795d4a89c1609646609d61e16
+      - type: file
+        url: https://raw.githubusercontent.com/feeddeck/feeddeck/main/app/linux/flatpak/app.feeddeck.feeddeck.svg
+        sha256: 4a73d70cdaa6fe230eb802fe0f5456df46dfe399826b044086ad46b74c2bd6ce
+      - type: archive
+        only-arches:
```
[the remark this answers is not in the document]
Comment (hfiguiere) [reply app.feeddeck.feeddeck.yml:36]: there is an aarch64 for once. (line 44)

**Comment by ricoberger** [comment]
Thanks everyone for the fast review process.

Do I have to do anything else, because I'm still getting a `404 - Page not found` error for https://flathub.org/apps/app.feeddeck.feeddeck?

**Comment by bbhtt** [comment]
Looks like the last official build failed due to GitHub issues, I triggered a rebuild in https://buildbot.flathub.org/#/builders/6/builds/71764, if successful it will be published in ~4-5 hours. You can also manage this by logging in there.

**Comment by bbhtt** [comment]
It's up now https://flathub.org/apps/app.feeddeck.feeddeck

**Comment by ricoberger** [comment]
Thank you 🙂

**Pull request closed without merging**